### ISSUE TITLE
Issue in number multiplication with 10, 20, etc

### ISSUE THREAD

**Issue by GolakiyaVishal** [dart-lang/sdk#56097]
In general math multiplication operation result is not as expected
Checked on [dartpad](https://dartpad.dev/) 
Image attached below.

While multiplying number is not giving proper result.
in some case we are simply trying to multiply with 10, 20, etc. but it is giving different result that expected.
Please check below sample multiplications.

11.41 * 10 = 114.1
1.41 * 10 = 14.1
0.41 * 10 = 4.1

11.42 * 10 = 114.2
1.42 * 10 = 14.2
0.42 * 10 = 4.2

11.43 * 10 = 114.3
**1.43 * 10 = 14.299999999999999**
0.43 * 10 = 4.3

**11.44 * 10 = 114.39999999999999**
**1.44 * 10 = 14.399999999999999**
0.44 * 10 = 4.4

11.45 * 10 = 114.5
1.45 * 10 = 14.5
0.45 * 10 = 4.5

**11.46 * 10 = 114.60000000000001**
1.46 * 10 = 14.6
**0.46 * 10 = 4.6000000000000005**

**11.46 * 10 = 114.60000000000001**
1.46 * 10 = 14.6
**0.46 * 10 = 4.6000000000000005**

**11.56 * 10 = 115.60000000000001
1.56 * 10 = 15.600000000000001
0.56 * 10 = 5.6000000000000005**

11.46 * 20 = 229.20000000000002
1.46 * 20 = 29.2
0.46 * 20 = 9.200000000000001

11.47 * 10.0 = 114.7
1.47 * 10.0 = 14.7
0.47 * 10.0 = 4.699999999999999

11.47 * 20.0 = 229.4
1.47 * 20.0 = 29.4
0.47 * 20.0 = 9.399999999999999

11.48 * 10.0 = 114.80000000000001
1.48 * 10.0 = 14.8
0.48 * 10.0 = 4.8

11.48 * 20.0 = 229.60000000000002
1.48 * 20.0 = 29.6
0.48 * 20.0 = 9.6

10.46 * 10.0 = 104.60000000000001
10.45 * 10.0 = 104.5
0.45 * 10.0 = 4.5
10.47 * 10.0 = 104.7
0.47 * 10.0 = 4.699999999999999
100.47 * 10.0 = 1004.7
10.48 * 10.0 = 104.80000000000001
0.48 * 10.0 = 4.8
100.48 * 10.0 = 1004.8000000000001

![image](https://github.com/dart-lang/sdk/assets/11039811/cb5ccce1-1573-43aa-9827-da1eab4f43e7)

**Comment by devoncarew** [comment]
I believe this is working as expected and is due to how numbers as represented as doubles. There's a bit more info on Dart's number semantics here: https://dart.dev/guides/language/numbers.

**Comment by GolakiyaVishal** [comment]
In that case, if it expected then below unit test case has to give success.

  test('simple test', (){
    expect(1.43 * 10, 14.3);
  });
  
  but it is failing ,
  
  is this same behavior, will find in other coding languages ?

**Comment by parlough** [comment]
Yes, consider checking out [What Every Programmer Should Know About Floating-Point Arithmetic](https://floating-point-gui.de/). The page on [comparisons](https://floating-point-gui.de/errors/comparison/) has some specific details about what you're seeing there.
